cost/kubecost/cluster/kubecost_cluster_rightsizing_recommendations.pt

```diff
@@ -0,0 +1,226 @@
+long_description ""
+severity "low"
+category "Cost"
+default_frequency "daily"
+tenancy "single"
+info(
+  version: "0.1",
+  provider: "Kubecost",
+  service: "Kubernetes",
+  policy_set: "Rightsize Clusters",
+  recommendation_type: "Usage Reduction"
+)
+
+###############################################################################
+# User inputs
+###############################################################################
+
+parameter "param_email" do
+  label "Email addresses"
+  description "Email addresses of the recipients you wish to notify"
+  type "list"
+end
+
+parameter "param_kubecost_host" do
+  type "string"
+  label "Kubecost Host"
+  description "Kubecost Host or IP Address of Kubecost LB"
+end
+
+parameter "param_min_nodes" do
+  type "number"
+  label "Minimum node count"
+  description "Minimum allowed node count in the cluster"
+  default 2
+end
+
+parameter "param_historical_days" do
+  type "number"
+  label "Number of days to analyze"
+  description "Historical range to analyze"
+  min_value 2
+  max_value 60
+  default 7
+end
+
+parameter "param_target_util" do
+  type "number"
+  label "Target Utilization (%)"
+  description "Desirable target utilization in percent"
+  default 80
+end
+
+parameter "param_recommendation_strategy" do
+  type "string"
+  label "Recommendation Strategy"
+  description "Recommendation strategy to apply"
+  allowed_values ["Single", "Multi", "Optimal"]
+end
+
+###############################################################################
+# Datasource
+###############################################################################
+
+datasource "ds_get_currency" do
+  request do
+    verb "GET"
+    scheme "http"
+    host $param_kubecost_host
+    path "/model/getConfigs"
+  end
+  result do
+    encoding "json"
+    field "currencyCode", jmes_path(response, "data.currencyCode")
+  end
+end
+
+datasource "ds_cluster_sizing" do
+  request do
+    run_script $js_cluster_sizing, $param_kubecost_host, $param_min_nodes, $param_historical_days, $param_target_util
+  end
+  result do
+    encoding "json"
+    field "clusterId", jmes_path(response, "data.parameters.clusterId")
+    field "clusterName", jmes_path(response, "data.parameters.clusterName")
+    field "totalNodeCount", jmes_path(response, "data.currentClusterInfo.totalCounts.totalNodeCount")
+    field "totalRAMGB", jmes_path(response, "data.currentClusterInfo.totalCounts.totalRAMGB")
+    field "totalVCPUs", jmes_path(response, "data.currentClusterInfo.totalCounts.totalVCPUs")
+    field "monthlyRate", jmes_path(response, "data.currentClusterInfo.monthlyRate")
+    field "recommendations", jmes_path(response, "data.recommendations")
+  end
+end
+
+script "js_cluster_sizing", type: "javascript" do
+  parameters "kubecost_host", "min_nodes", "historical_days", "target_util"
+  result "request"
+  code <<-EOS
+    var request = {
+      "verb": "GET",
+      "scheme": "http",
```

Comment:
Question for my information - How come the scheme is HTTP rather than HTTPS?

Reply:
Well, we have a kubecost API that uses HTTP protocol. I was testing with HTTPS and our API calls wouldn't work. On the other hand, if you use HTTP scheme in the policy and you try to call the kubecost demo API that uses HTTPS, it does work.

I know for many reasons probably HTTPS is the way to go, so I'll be in touch to see if we change the value in this line.